nsv13/code/modules/overmap/weapons.dm

```diff
@@ -155,130 +133,90 @@
 
 /obj/structure/overmap/proc/swap_to(what=FIRE_MODE_PDC)
-	switch(what)
-		if(FIRE_MODE_PDC)
-			fire_delay = initial(fire_delay) //all ships start on PDC mode
-			weapon_range = initial(weapon_range)
-			fire_mode = FIRE_MODE_PDC
-		if(FIRE_MODE_RAILGUN)
-			fire_delay = 10 //Very limited ammo
-			weapon_range = initial(weapon_range)+30 //Gain a large range bonus. This will take care of a lot of combat.
-			fire_mode = FIRE_MODE_RAILGUN
-		if(FIRE_MODE_TORPEDO)
-			fire_delay = 5 //These things rip into your hull, but can be easily shot down
-			weapon_range = initial(weapon_range)+30 //Most combat takes place at extreme ranges, torpedoes allow for this.
-			fire_mode = FIRE_MODE_TORPEDO
+	if(ai_controlled || (!linked_areas.len && !main_overmap)) //AI ships and fighters don't have interiors
+		if((what == FIRE_MODE_TORPEDO) && !torpedoes) //Out of torpedoes
+			return FALSE
+		if((mass < MASS_MEDIUM) && (what > FIRE_MODE_TORPEDO)) //Little ships don't have railguns or lasers
+			return FALSE
+	else if(!weapons || !weapons[what] || !weapons[what].len) //Hero ship doesn't have any weapons of this type
+		return FALSE
+
+	var/datum/ship_weapon/SW = weapon_types[what]
+	fire_delay = initial(fire_delay) + SW.fire_delay
+	weapon_range = initial(weapon_range) + SW.fire_delay
+	fire_mode = what
 	if(ai_controlled)
 		fire_delay += 10 //Make it fair on the humans who have to actually reload and stuff.
 
-/obj/structure/overmap/proc/fire_railgun(atom/target)
-	if(ai_controlled) //AI ships don't have interiors
-		fire_lateral_projectile(/obj/item/projectile/bullet/railgun_slug, target, 10)
-		return
-	var/proj_type = null //If this is true, we've got a launcher shipside that's been able to fire.
-	var/atom/proj_object = null
-	var/fired = FALSE
-	for(var/X in railguns)
-		if(istype(X, /obj/structure/ship_weapon/railgun))
-			var/obj/structure/ship_weapon/railgun/RG = X
-			if(RG.can_fire())
-				proj_object = RG.fire()
-				if(!proj_object)
-					continue
-				if(istype(proj_object, /obj/item/twohanded/required/railgun_ammo))
-					var/obj/item/twohanded/required/railgun_ammo/RA = proj_object
-					proj_type = RA.proj_type
-					qdel(proj_object)
-				if(proj_type)
-					fired = TRUE
-					break
-	if(!fired)
-		to_chat(gunner, "<span class='warning'>DANGER: Launch failure! Railgun systems are not loaded.</span>")
-		return
-
-
-	var/sound/chosen ='nsv13/sound/effects/ship/railgun_fire.ogg'
-	relay_to_nearby(chosen)
-	flick("railgun_charge",railgun_overlay)
-	shake_everyone(2)
-	if(proj_type)
-		fire_lateral_projectile(proj_type, target, 5)
+	return TRUE
 
-/obj/structure/overmap/proc/shake_everyone(severity)
-	for(var/mob/M in mobs_in_ship)
-		if(M.client)
-			shake_camera(M, severity, 1)
+/obj/structure/overmap/proc/fire_weapon(atom/target, mode=fire_mode, lateral=TRUE) //"Lateral" means that your ship doesnt have to face the target
+	if(ai_controlled || (!linked_areas.len && !main_overmap)) //AI ships and fighters don't have interiors
```

Comment:
This means that if the nostromo fires powerful weapons it wont shake.

Reply:
The Nostromo has linked areas actually